### ISSUE TITLE
Dirs/files are missing when scanning on windows after 1.27.12

### ISSUE THREAD

**Issue by ovitemp** [syncthing/syncthing#9859]
### What happened?

I use installed SyncTrayzor and setup send-receive drive D: on 2 computers over the internet. I noticed missing files on one computer and after some tests I found that some folders are not even scanned.
I completely removed SyncTrayzor and folder containing the deployed Syncthing and reinstalled. I tested few times and every time after adding the D: drive as shared folder, without even share it at all, SyncTrayzor reported Global and Local State only 405 folders and a number of files.
I found that replacing syncthing 1.28.1 with older version 1.27.12 resulted in SyncTrayzor reporting 
Global State   Files 65,641 Folders 2,107  ~413 GiB
that means 4 times more folders. I am not sure that all folders are detected, but looks much near of the expected size.
If necessary please let me know if I must upload any kind of logs.
Regards,
Ovi

### Syncthing version

v1.28.1

### Platform & operating system

Windows 10 64bit

**Comment by calmh** [comment]
Anything funky with that disk; deduplication, compression, etc?

**Comment by hujun-open** [comment]
I have same issue, I have a folder around 100GB, 300k files,  66k sub-folders; a small part of sub-folders/files are not scanned, the weird thing is it works if I add missed sub-folder as separate syncthing folder.  I have checked the log and debug, can't find any error msg 

I am using v1.28.1 and windows 11

**Comment by calmh** [comment]
Anything funky with that disk; deduplication, compression, etc?

**Comment by hujun-open** [comment]
> Anything funky with that disk; deduplication, compression, etc?

I am using windows 10 pro, I don't think there is deduplication since it is a windows server feature, there is also no compression; there is bitlocker on the drive though

**Comment by hujun-open** [comment]
I also tried 1.27.12, it seems all the folders missed by 1.28.1 could be scanned by 1.27.12

**Comment by rasa** [comment]
Are there directory junctions or symlinks involved? What about network shares? Are you sure you are using 1.28.1, and not 1.28.0?

**Comment by hujun-open** [comment]
there is no symbol link or network share in my case, and yes, it was 1.28.1

**Comment by hakakou** [comment]
Thank goodness I found this thread as I was going nuts. So started noticing that files where missing on the laptop. 

Anyway I have tryied deleting the index-v0.14.0.db, but no result. I have tried to completely remove the folder from synthing and recreate a new share... but still entire root level folders are simply ignored. Here are some screenshots what I see:

So I have this folder
![image](https://github.com/user-attachments/assets/33365c8d-cb2e-42a5-a323-3ca32525e09d)

 It's completely new and unshared:
![image](https://github.com/user-attachments/assets/dd972d2f-a1a4-42eb-85be-2eb1a230d749)

But look at this:
![image](https://github.com/user-attachments/assets/c976e7ec-542c-4193-8a1a-c0ceca2a9b86)
-----
![image](https://github.com/user-attachments/assets/7d538e4c-62e4-4ab9-b023-5e25c7756625)
-----
![image](https://github.com/user-attachments/assets/3b861ccc-635f-4ab8-9c70-4524b8002d07)
-----
![image](https://github.com/user-attachments/assets/b17fc836-5589-42af-97ff-0f46fed807e2)
-----
Is this normal? I tested each and every top-level folder in **postman**:

Applications\   -------------------> OK
Archives\   -------------------> OK
Books\   -------------------> OK
Certificates\   -------------------> OK
Config\   -------------------> OK
Delphi\   -------------------> OK
Deployments\   -------------------> OK
Documents\     -------------------> OK
Ideas\     -------------------> OK
Links\    -------------------> OK
Net\   ------------------->  OK
NodeJS\   ---------------> "No such object in the index"
OpenSSL\  ---------------> "No such object in the index"
Papers\    ---------------> "No such object in the index"
PHP\   ---------------> "No such object in the index"
PowerShell\  ---------------> "No such object in the index"
Python\  ---------------> "No such object in the index"
Remote Desktop\   ---------------> "No such object in the index"
Tests\   ---------------> "No such object in the index"
Tools\   ---------------> "No such object in the index"
Work\   ---------------> "No such object in the index"

Do you see a pattern? It's like the folder was scanned alphabetically, and literaly stopped in the middle! Even If try deleting the index-v0.14.0.db, deleting the folder share itself the same happens!. Of course these folder and all their contents don't sync to the laptop.

**Comment by calmh** [comment]
Yeah. Are you up for trying the latest nightly build which may affect this? 

https://syncthing-apt.s3.fr-par.scw.cloud/nightly/syncthing-windows-amd64-v1.29.0-rc.1.dev.12.g79ef57d0.zip

If it helps, a scan with `scanner` debugging enabled should yield one or more lines like `Skipping non-regular file ...` which would be interesting to see.

If it doesn't, I know where to hook in more debugging now.

**Comment by hakakou** [comment]
I din't see your reply, but before I did a search for directory junctions like this:

XXX>dir /AL /S
Volume in drive C is Windows 11
Directory of XXX\node_modules\@yourorg

04/12/2024  22:26    <JUNCTION>     react-api-client [XXX]
...
               
surely enought after deleting these directory junctions (mostly created in node_modules folders by npm) the sync seems to work.

**Comment by calmh** [comment]
junctions ❤️ 

Those should ideally already be handled by another code path iirc, but 🤷

**Comment by hakakou** [comment]
Is there any plan to fix this? I discovered now that thousands of files were not syncing to my laptop because of this.

**Comment by calmh** [comment]
I mean, I posted a link to possibly fixed build for you to test and report back.

**Comment by hujun-open** [comment]
> Yeah. Are you up for trying the latest nightly build which may affect this?
> 
> https://syncthing-apt.s3.fr-par.scw.cloud/nightly/syncthing-windows-amd64-v1.29.0-rc.1.dev.12.g79ef57d0.zip
> 
> If it helps, a scan with `scanner` debugging enabled should yield one or more lines like `Skipping non-regular file ...` which would be interesting to see.
> 
> If it doesn't, I know where to hook in more debugging now.

This build works for me, it turns out I do have some symlink/junction that I didn't know of

**Comment by hujun-open** [comment]
> > Yeah. Are you up for trying the latest nightly build which may affect this?
> > https://syncthing-apt.s3.fr-par.scw.cloud/nightly/syncthing-windows-amd64-v1.29.0-rc.1.dev.12.g79ef57d0.zip
> > If it helps, a scan with `scanner` debugging enabled should yield one or more lines like `Skipping non-regular file ...` which would be interesting to see.
> > If it doesn't, I know where to hook in more debugging now.
> 
> This build works for me, it turns out I do have some symlink/junction that I didn't know of

if the fix is skipping the symlink/junction, then it would nicer to show skipped file in web UI (or at least in the log without enabling the debug)

**Comment by ovitemp** [comment]
I just came back from a long trip and unfortunately I forgot completely about this problem.
Yes, indeed, I have a disk accessible through a "path" in computer management, not as a letter, dir /AL /S comes out with 3 Junctions because I have other 2 disks that I plug them from time to time.
So as I understand, for now there is no release that fix this problem, only the nightly build, right?

**Comment by acolomb** [comment]
Please test with the current release v1.29.2. It should be fixed since v1.29.1 already.

**Comment by calmh** [comment]
This was indeed fixed in 1.29.1, I just forgot to link the issue as there were duplicates.
